### ISSUE TITLE
TextControl doesn't allow `date`, `time`, `datetime-local` types

### ISSUE THREAD

**Issue by cawa-93** [WordPress/gutenberg#59544]
### Description

![зображення](https://github.com/WordPress/gutenberg/assets/1662812/3c67263e-f8e6-4881-8a6a-d049c709d470)

In fact, this field is fully functional (for the web), but it raises a typescript error.

### Step-by-step reproduction instructions

Code with issue:
```ts
<TextControl
    label={__('Start Date', 'road')}
    type="datetime"
/>;
```

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
Thanks for the report.

This type restriction was added in #45433, but since it is just an annotation, it seems possible to specify types that are not allowed.  I think this issue can be considered an improvement that expands the available types.

P.S. `datatime` does not seem to exist as [a type that can be used in the input element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types).

**Comment by mirka** [comment]
Yes, it seems reasonable to allow these date/time types.

**Comment by dphyled** [comment]
Dibs! :D

**Comment by mirka** [comment]
Closed by #59666
